include/bitcoin/server/web/socket.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_SERVER_WEB_SOCKET_HPP
+#define LIBBITCOIN_SERVER_WEB_SOCKET_HPP
+
+#include <cstdint>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <bitcoin/protocol.hpp>
+#include <bitcoin/server/define.hpp>
+#include <bitcoin/server/settings.hpp>
+#include <bitcoin/server/messages/message.hpp>
+
+// TODO: avoid public exposure of this header (move types to cpp).
+#include <../src/web/external/mongoose/mongoose.h>
```

Comment:
What's the plan here?

Reply:
None at the moment.  I thought by moving the include outside of "include/bitcoin/server/..." is solving "avoiding the public exposure".  I left the comment in because I'm not sure if you're asking for a more involved re-factor of their _all-in-one 6K line_ header file?  If so, I need more detail of what you had in mind.  The all in one file is more or less a concatenation of all of their individual headers (taking care for ordering/deps).

Reply:
I did this with zeromq. The libbitcoin developer does not need the zeromq headers when linking to a libbitcoin-server or libbitcoin-client library. Those libs only include zeromq in their cpp files. I accomplished this by redefining one (I think it was just one) of the zeromq structures in one of our headers. I don't see a large dependency on mongoose types here so I'm hopeful the same can be accomplished.